composer/core/state.py

```diff
@@ -618,8 +618,9 @@ def _dataset_state_dict(self) -> Dict[str, Any]:
 
         dataset = self._dataset_of(self.train_dataloader)
         if hasattr(dataset, 'state_dict'):
-            sample = int(self.timestamp.sample_in_epoch)
-            obj['train'] = dataset.state_dict(sample)  # pyright: ignore
+            num_samples = int(self.timestamp.sample_in_epoch)
+            from_beginning = True
+            obj['train'] = dataset.state_dict(num_samples, from_beginning)  # pyright: ignore
```

Comment:
Maybe have this `obj['train'] = dataset.state_dict(num_samples, True)` ? and add a comment that it used in `StreamingDataset` ??